### ISSUE TITLE
Restore deleted top tiles

### ISSUE THREAD

**Issue by VideoPlayerCode** [brave/browser-android-tabs#2357]
I have searched and cannot find any related ticket for this problem:

The top tiles (automatic shortcuts to frequently visited placed) permanently blacklist sites if you remove their tile. I accidentally removed YouTube a month ago and cannot get it back despite visiting that site daily.

1. Tap and hold on a top tile site
2. Choose Remove
3. The site will never reappear, which is good and expected. But there is seemingly NO WAY to re-add/allow the deleted site again.

**Comment by srirambv** [comment]
Tiles are based on history. There is no way to edit and add them. The (only easy) way to add on top tile would be to delete all of them and visit YT a few times to get that added as a tile. If you do visit other sites it may get added before YT does. So after removing all tiles you may need to visit YT a bunch to see that added as top tile. 

Related issue for adding custom user favourites https://github.com/brave/brave-browser/issues/7258

**Comment by VideoPlayerCode** [comment]
@srirambv Thanks a lot for your answer, Sriram! But here are my top sites. Only two entries. It has been like this for a month and YouTube is not coming back despite me almost only browsing that site.

Will deleting the remaining top tiles force Brave to clear all data and learn YouTube again?

https://i.imgur.com/fElZgoD.png

**Comment by VideoPlayerCode** [comment]
I deleted all remaining tiles but so far it isn't suggesting any sites at all. I really hope deleting all tiles hasn't simply added even more sites to the "don't show as tile anymore" blacklist. :-P

**Comment by srirambv** [comment]
With the new background image on new tab you would only see 4 tiles at a time. You would need to disable the image to get more tiles. 

cc: @anthonypkeane

**Comment by VideoPlayerCode** [comment]
@srirambv Yep. But I have 0 tiles. Every time it suggests a new top site tile, I delete it, hoping YouTube is gonna come up, but it never does.

So... the thing you proposed has just made me blacklist even more sites. You said that Brave would reset itself and start suggesting new top tiles again if I delete all tiles, but that is not the case, hehe. Now everything is blacklisted. YouTube and all other sites. ;P

Seems like only solution is to reinstall Brave for Android.

**Comment by srirambv** [comment]
@VideoPlayerCode There is no concept of blacklisting. It is just removed and would get added back. Another workaround to get toptiles back is to clear browsing data and cache. This would reset to default toptiles and as you browser you should be able to get back the ones you need. Do note it will clear any sessions that are valid if you do choose to clear browsing data.

**Comment by VideoPlayerCode** [comment]
@srirambv Oh. I assumed there was a blacklist since I use YouTube for like 2 hours a day and barely visit any other sites, but after a month it still hadn't suggested YouTube. It is weird.

I did add YouTube to my bookmarks to compensate for the lost top tile. Could that be interfering and preventing the site from appearing again? 

Thanks for your other suggestions, I will investigate them soon!

**Comment by VideoPlayerCode** [comment]
Yesterday I deleted my YouTube bookmark (regular Brave bookmark). The top tiles are still not suggesting YouTube. But it is suggesting random sites I give 1 visit to today. I am now 100% sure that Brave had a top tile blacklist that refuses to show any URL that you have manually removed from top tiles.

Will be erasing the app's data and cache as you suggested. Basically the same as a reinstall of Brave since every single setting will be lost. But it's the only solution. 

The planned configurable top/favorites feature is sorely needed, hehe.

**Comment by VideoPlayerCode** [comment]
Hehe "told you so":

https://github.com/brave/browser-android-tabs/blob/d5a126abe14a58056e85431337884cd4f5d3d217/chrome/browser/android/explore_sites/most_visited_client.h#L12-15

**Comment by VideoPlayerCode** [comment]
I see now that there are APIs for clearing the "top tiles" blacklist. Could you please add a sn advanced menu setting to clear that blacklist?
